tests/devices/qubit/test_simulate.py

```diff
@@ -71,6 +71,33 @@ def compute_matrix():
     assert qml.math.allclose(result, -1.0)
 
 
+# pylint: disable=too-few-public-methods
+class TestSparsePipeline:
+    """System tests for the sparse pipelines."""
+
+    ground_state = np.array([1, 0, 0, 0, 0, 0, 0, 0])
+    cat_state = np.array([1, 0, 0, 0, 0, 0, 0, 1]) / np.sqrt(2)
+
+    @pytest.mark.parametrize(
+        "state",
+        [
+            ground_state,
+            cat_state,
+        ],
+    )
+    def test_sparse_op(self, state):
+        """Test that a sparse QubitUnitary operation works on default.qubit with expval measurement."""
+        mat = sp.sparse.csr_matrix([[0, 1], [1, 0]])
+
+        @qml.qnode(qml.device("default.qubit"))
+        def circuit(state, mat):
+            qml.StatePrep(state, wires=range(8), pad_with=0)
+            qml.QubitUnitary(mat, wires=[0])
+            return qml.expval(qml.Z(0))
```

Comment:
Since this is in `test_simulate`, can we directly pass a `QuantumScript` into `simulate`? I'm not sure we need to couple this test to the rest of the workflow infrastructure.